Software/src/MoodLampManager.cpp

```diff
@@ -167,8 +169,17 @@ void MoodLampManager::updateColors(const bool forceUpdate)
 	DEBUG_MID_LEVEL << Q_FUNC_INFO << newColor.rgb();
 
 	bool changed = (m_lamp ? m_lamp->shine(newColor, m_colors) : false);
-	if (changed || !m_isSendDataOnlyIfColorsChanged || forceUpdate)
+	if (changed || !m_isSendDataOnlyIfColorsChanged || forceUpdate) {
 		emit updateLedsColors(m_colors);
+		if (forceUpdate) {
+			emit moodlampFrametime(1000);
+			m_elapsedTimer.restart();
```

Comment:
Should you reset m_frames here too?

Reply:
You are right. It looked like I was getting expected results so I did not look further at the time, but turns out `m_isSendDataOnlyIfColorsChanged` wasn't being updated while the app was running which messed with my tests.